include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -337,12 +337,46 @@ MatrixFree<dim, Number, VectorizedArrayType>::internal_reinit(
             Utilities::MPI::this_mpi_process(task_info.communicator);
           task_info.n_procs =
             Utilities::MPI::n_mpi_processes(task_info.communicator);
+
+          Assert(additional_data.communicator_sm == MPI_COMM_SELF,
+                 ExcNotImplemented());
+
+#ifndef DEAL_II_WITH_MPI
+          Assert(false, ExcInternalError());
+#else
+          if (true)
+            {
+              task_info.communicator_sm = additional_data.communicator_sm;
+            }
+#  if DEAL_II_MPI_VERSION_GTE(3, 0)
+          else if (true)
+            {
+              MPI_Comm comm_sm;
+              MPI_Comm_split(task_info.communicator,
+                             task_info.my_pid == 2,
+                             task_info.my_pid,
+                             &comm_sm);
```

Comment:
Now we never go into this part, but I am confused by this statement and `my_pid == 2` in particular. Can we maybe postpone this part until we have a reasonable criterion here? Or is it just me not understanding the `Comm_split` context here?

Reply:
I think I wanted to write `task_info.my_pid / 2`. In the future the communicator should come from outside.

But anyhow, what I wanted to do is to run all tests with pure MPI, pure SM-MPI and hybrid-MPI. Without having to modify all tests this is the place where it can be achieved without too much work.

Reply:
I see, I guess we can keep it for now and simply delete it once everything else is in place.

Reply:
I will remove this in the final version.